apm-agent-core/src/main/java/co/elastic/apm/agent/impl/stacktrace/StacktraceConfiguration.java

```diff
@@ -88,7 +103,17 @@ public int getStackTraceLimit() {
         return stackTraceLimit.get();
     }
 
-    public long getSpanFramesMinDurationMs() {
-        return spanFramesMinDurationMs.getValue().getMillis();
+    public long getSpanStackTraceMinDurationMs() {
+        if (spanStackTraceMinDurationMs.isDefault()) {
+            long spanFramesMinDurationMsValue = spanFramesMinDurationMs.getValue().getMillis();
+            if (spanFramesMinDurationMsValue == 0) {
+                return -1;
+            } else if (spanFramesMinDurationMsValue < 0) {
+                return 0;
+            } else {
+                return spanFramesMinDurationMsValue;
```

Comment:
One last change needed, if spanFramesMinDurationMsValue is also default, we want to return the spanStackTraceMinDurationMs default, not the spanFramesMinDurationMsValue default (doesn't matter now, but defaults can change)

Reply:
Hi @jackshirazi , please check last commit